### ISSUE TITLE
Dmenu, despite appearing to try to display on my first monitor, does not actually appear on it

### ISSUE THREAD

**Issue by happysmash27** [WayfireWM/wayfire#236]
In my second monitor, text continues as if going from the first one, when running dmenu, but dmenu does not actually appear in my first monitor. Monitor configuration: 

```
[DP-3]
# position of top-left corner in global compositor space
layout = 0@0
#add a custom mode, can be used in the mode option
#custom_mode_1 = 74.50  1280 1344 1472 1664  720 723 728 748 -hsync +vsync
#set resolution
#mode = 1280x720@60.000
# set output scale
scale = 1.00
#set rotation
transform = normal

[HDMI-A-1]
layout = 1280@0
scale = 1.00
transform = normal
```

**Comment by soreau** [comment]
I don't really understand what you mean here, can you explain a bit more and perhaps provide a screenshot if it will help?

**Comment by happysmash27** [comment]
![brokendmenu](https://user-images.githubusercontent.com/3620657/59142131-e355c380-896d-11e9-99d7-e8a577c72c41.png)

**Comment by soreau** [comment]
That's much clearer, thanks.

**Comment by ammen99** [comment]
You can use bemenu: https://github.com/Cloudef/bemenu
It is very similar to dmenu and is wayland-native, so it works much better.

The problem with dmenu is that it doesn't know which output is active. It looks into the lastly focused Xwayland window, or picks an arbitrary output. However, if the last focused window is a Wayland window, dmenu can't know this and opens on a different output. And since that output is not focused by wayfire, dmenu doesn't get input focus.

**Comment by ammen99** [comment]
As @soreau said on IRC, seems an issue like described here: https://faq.i3wm.org/question/4982/dmenu-multi-monitor-issue.1.html

TLDR: update dmenu.

But seriously, use wayland-native tools, good alternatives are already available. (bemenu, dmenu-wl, etc).
